### PR TITLE
Update SKTouchHandler.cs to fix crash bug caused by passing incorrect parameter, added pressure and eraser handling

### LINKED ISSUE #1211: [BUG] Crash bug in SKTouchHandler.OnTouch
**Description**

https://github.com/mscherotter/SkiaSharp/blob/master/source/SkiaSharp.Views.Forms/SkiaSharp.Views.Forms.Android/SKTouchHandler.cs

    var toolType = evt.GetToolType(id); 

in line 50 incorrectly passes the pointerID, not the pointer index and causes an exception: pointerIndex out of range

**Code**
The line should be 

    var toolType = evt.GetToolType(pointer);

**Expected Behavior**

No exception

**Actual Behavior**

An exception when touching the canvas
**Basic Information**

- Version with issue:  1.68.2-preview.50
- Last known good version:  1.68.2-preview.45
- IDE:   Visual Studio 
- Platform Target Frameworks: Android
  - Android: 9.0 
- Target Devices:   
  - 
  - 

**Screenshots**

**Reproduction Link**

### PR DESCRIPTION
**Pull request by mscherotter**
Fixed crash bug where id was incorrectly passed instead pointer to evt.GetToolType().

**Description of Change**

- Changed parameter from id to pointer - bug introduced in previous pull request.
- Added pressure parameter for touch handling
- Added button handling to enable eraser in stylus input

**Bugs Fixed**

- Fixes #1211 

**API Changes**
- SkTouchEventArgs.Pressure float parameter has been added
- Constructor for SKTouchEventArgs() with overloaded pressure parameter

**PR Checklist**

- [ ] Has tests (if omitted, state reason in description)
- [ ] Rebased on top of master at time of PR
- [ ] Changes adhere to coding standard
- [ ] Updated documentation

### CODE REVIEW

File: source/SkiaSharp.Views.Forms/SkiaSharp.Views.Forms.Shared/SKTouchEventArgs.cs [hunk 1]
```diff
@@ -42,9 +42,11 @@ public SKTouchEventArgs(long id, SKTouchAction type, SKMouseButton mouseButton,
 
 		public int WheelDelta { get; private set; }
 
+		public float Pressure { get; internal set; } = 1.0f;
```
Comment (mattleibow) [inline source/SkiaSharp.Views.Forms/SkiaSharp.Views.Forms.Shared/SKTouchEventArgs.cs:45]: Maybe add an overload for this in the constructor so this args can be created from outside SkiaSharp.

File: source/SkiaSharp.Views.Forms/SkiaSharp.Views.Forms.Android/SKTouchHandler.cs [hunk 2]
```diff
@@ -47,15 +47,32 @@ private void OnTouch(object sender, View.TouchEventArgs e)
...
+
+			if (toolType == MotionEventToolType.Eraser)
+			{
+				button = SKMouseButton.Middle;
+			}
+			else if (evt.ButtonState == MotionEventButtonState.StylusSecondary)
+			{
+				button = SKMouseButton.Right;
+			}
+
```
Comment (mattleibow) [inline source/SkiaSharp.Views.Forms/SkiaSharp.Views.Forms.Android/SKTouchHandler.cs:65]: Maybe move this into a separate method to be like the others

File: source/SkiaSharp.Views.Forms/SkiaSharp.Views.Forms.Shared/SKTouchEventArgs.cs (hunk 1 above)
Comment (mscherotter) [reply source/SkiaSharp.Views.Forms/SkiaSharp.Views.Forms.Shared/SKTouchEventArgs.cs:45]: will do

File: source/SkiaSharp.Views.Forms/SkiaSharp.Views.Forms.Android/SKTouchHandler.cs (hunk 2 above)
Comment (mscherotter) [reply source/SkiaSharp.Views.Forms/SkiaSharp.Views.Forms.Android/SKTouchHandler.cs:65]: will do

**Comment by mattleibow** [comment]
Thanks for the fix and the pressure feature. Great addition!

**Pull request merged**